src/qibo/tomography/gate_set_tomography.py

```diff
@@ -0,0 +1,277 @@
+from functools import cache
+from inspect import signature
+from itertools import product
+from typing import Union
+
+import numpy as np
+from sympy import S
+
+from qibo import Circuit, gates, symbols
+from qibo.backends import GlobalBackend
+from qibo.config import raise_error
+from qibo.hamiltonians import SymbolicHamiltonian
+
+
+@cache
+def _gates(nqubits):
+    return list(
+        product(
+            [(gates.I,), (gates.X,), (gates.H,), (gates.H, gates.S)], repeat=nqubits
+        )
+    )
+
+
+@cache
+def _measurements(nqubits):
+    return list(product([gates.Z, gates.X, gates.Y, gates.Z], repeat=nqubits))
+
+
+@cache
+def _observables(nqubits):
+    return list(product([symbols.I, symbols.Z, symbols.Z, symbols.Z], repeat=nqubits))
+
+
+@cache
+def _get_observable(j, nqubits):
+    if j == 0 and nqubits == 3:
+        raise ValueError("Invalid parameters: j=0 and nqubits=3")
```

Comment:
I did not write the function `_get_observable()`. Not sure who split up the function `_expectation_value()` to create a new function called `_get_observable()`.

Reply:
I figured this doesn't add any value to the code. I've removed the nqubits==3 and the sympy lines.

Reply:
I still see the 
``` python
if j == 0 and nqubits >= 3:
```
check. I would change this to simply:
``` python
if not nqubits in (1,2):
```
as in the other functions, since these are the only two values of `nqubits` we support anyway and in those cases `j==0` doesn't represent a problem.